framework/src/userobject/ElementQualityChecker.C

```diff
@@ -0,0 +1,201 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#include "ElementQualityChecker.h"
+#include "MooseError.h"
+
+#include "libmesh/elem_quality.h"
+#include "libmesh/enum_elem_quality.h"
+
+MooseEnum
+ElementQualityChecker::QualityMetricType()
+{
+  return MooseEnum("ASPECT_RATIO SKEW SHEAR SHAPE MAX_ANGLE MIN_ANGLE CONDITION DISTORTION TAPER "
+                   "WARP STRETCH DIAGONAL ASPECT_RATIO_BETA ASPECT_RATIO_GAMMA SIZE JACOBIAN");
+}
+
+MooseEnum
+ElementQualityChecker::FailureMessageType()
+{
+  return MooseEnum("WARNING ERROR", "WARNING");
+}
+
+template <>
+InputParameters
+validParams<ElementQualityChecker>()
+{
+  InputParameters params = validParams<ElementUserObject>();
+  params.addClassDescription("Class to check the quality of each element using different metrics "
+                             "from libmesh.");
+
+  params.addRequiredParam<MooseEnum>("metric_type",
+                                     ElementQualityChecker::QualityMetricType(),
+                                     "Type of quality metric to be checked");
+  params.addParam<Real>("upper_bound", "the upper bound for provided metric type");
+  params.addParam<Real>("lower_bound", "The lower bound for provided metric type");
+  params.addParam<MooseEnum>("failure_type",
+                             ElementQualityChecker::FailureMessageType(),
+                             "The way how the failure of quality metric check should respond");
+  params.set<MultiMooseEnum>("execute_on") = "initial";
+
+  return params;
+}
+
+ElementQualityChecker::ElementQualityChecker(const InputParameters & parameters)
+  : ElementUserObject(parameters),
+    _m_type(getParam<MooseEnum>("metric_type").getEnum<ElemQuality>()),
+    _has_upper_bound(isParamValid("upper_bound")),
+    _has_lower_bound(isParamValid("lower_bound")),
+    _upper_bound(_has_upper_bound ? getParam<Real>("upper_bound") : 0.0),
+    _lower_bound(_has_lower_bound ? getParam<Real>("lower_bound") : 0.0),
```

Comment:
Why not just use default values in the param object?

Reply:
@permcody You're correct, that would be better. Handling the defaults is a little tricker than that here, though. Every one of these metrics has different default thresholds per element type in libmesh. I think the right way to handle this is to defer to those defaults if 'upper_bound' or 'lower_bound' is not specified by the user, but use the user-supplied values if they exist.

Reply:
You need to initialize all of your Reals in the constructor too. I know this may seem redundant since you reset them in initialize, but this is just good defensive coding. If your UO gets turned off and initialize never gets called but for some reason another public method does to retrieve those values, they'd be in an invalid state. Just get in the habit of initializing all POD types.

Reply:
Oh, Since you are intentionally not using param defaults. It might be worth putting in a note as to why just so it's clear to another developer.